### PR TITLE
[py2py3] Migrate WMCore/WMFactory

### LINKED ISSUE #9970: [py2py3] Migrate WMCore/WMFactory
**Impact of the new feature**
This involves migrating `WMCore/WMFactory`. It does not depend on any other wmcore module, so it can be tackled straight away.

This change can directly impact

```
'WMComponent_TaskArchiver',
'WMComponent_RetryManager',
'WMCore_ThreadPool',
'WMCore_BossAir',
'WMCore_JobSplitting',
'WMCore_WMInit',
'WMCore_WebTools',
'WMComponent_DBS3Buffer',
'WMCore_ProcessPool',
'WMCore_WorkQueue',
'WMCore_ReqMgr',
'WMCore_WMSpec',
'WMCore_Storage',
'WMComponent_AnalyticsDataCollector',
'WMCore_WMRuntime'
```

since these modules import `WMCore/WMFactory`.

**Is your feature request related to a problem? Please describe.**
Make WMCore/WMFactory compatible with both python2 and python3.

**Describe the solution you'd like**
Run futurize on WMCore.WMFactory, then apply manual changes if needed

### PR DESCRIPTION
**Pull request by mapellidario**
Fixes #9970

#### Status
not-tested

#### Description
Run futurize on WMCore/WMFactory

#### Is it backward compatible (if not, which system it affects?)
It should be. However, any bug would affect directly

```
'WMComponent_TaskArchiver',
'WMComponent_RetryManager',
'WMCore_ThreadPool',
'WMCore_BossAir',
'WMCore_JobSplitting',
'WMCore_WMInit',
'WMCore_WebTools',
'WMComponent_DBS3Buffer',
'WMCore_ProcessPool',
'WMCore_WorkQueue',
'WMCore_ReqMgr',
'WMCore_WMSpec',
'WMCore_Storage',
'WMComponent_AnalyticsDataCollector',
'WMCore_WMRuntime'
```

#### External dependencies / deployment changes
requires python-future

### CODE REVIEW

**Comment by amaltaro** [comment]
@mapellidario I assume this one is ready for a review, however, in the future please remove the "Do not merge yet" flag when it's ready from your side.

**Comment by mapellidario** [comment]
Thanks Alan! I am sorry for not having updated the label. I opened the the PR, but then got dragged away with the preparation of the presentation. It seemed like a straight-forward change, but the last time we  imported `object` from future's builtins we completely broke JSONThunker, so I was a bit worried by this PR and did not call it as complete as soon as the jenkins results came in. 
I wanted to have another look, but what could I check further? Maybe nothing at this moment, we will have to deploy these changes in a testbed. 

TL;DR Thanks for merging this while I was scaried by the last time we made such a change!

**Comment by amaltaro** [comment]
This is now live in testbed with "   *   WMCore 1.4.2.pre2 release candidate for HG2011"
I will start a few tests today/tomorrow and bug you if I find problems, it looks safe to me though.
